addon/components/mapbox-gl.js

```diff
@@ -41,53 +40,53 @@ export default Component.extend({
   initOptions: null,
 
   /**
-    An action function to call when the map has finished loading. Note that the component does not yield until the map has loaded, 
+    An action function to call when the map has finished loading. Note that the component does not yield until the map has loaded,
     so this is the only way to listen for the mapbox load event.
 
     @argument mapLoaded
     @type {Function}
   */
   mapLoaded: noop,
 
-  init() {
-    this._super(...arguments);
-
-    this.map = null;
-    this.glSupported = MapboxGl.supported();
-
-    const mbglConfig = getOwner(this).resolveRegistration('config:environment')['mapbox-gl'] || {};
-    warn('mapbox-gl config is missing in config/environment', mbglConfig, { id: 'ember-mapbox-gl.config-object' });
-    warn('mapbox-gl config is missing an accessToken string', typeof mbglConfig.accessToken === 'string', { id: 'ember-mapbox-gl.access-token' });
-
-    MapboxGl.accessToken = mbglConfig.accessToken;
-  },
+  _loader: MapboxLoader,
+  _isMapLoaded: false,
+  _glSupported: computed('_loader.Mapbox', function() {
+    const Mapbox = get(this, '_loader.Mapbox');
+    if (Mapbox) {
+      return Mapbox.supported();
+    }
+  }).readOnly(),
+
+  _map: computed('_loader.Mapbox', 'element', function() {
+    const Mapbox = get(this, '_loader.Mapbox');
+    const { element } = this;
+    if (Mapbox && element) {
+      const config = get(getOwner(this).resolveRegistration('config:environment'), 'mapbox-gl.map');
+      const options = assign({}, config, get(this, 'initOptions'));
+      options.container = element;
+
+      const map = new Mapbox.Map(options);
+      map.once('load', bind(this, this._onLoad, map));
+      return map;
+    }
+  }).readOnly(),
```

Comment:
Cool! I have been subclassing this a lot, but I don't think anything should be affected. The move to indicate these as private is good for that.